flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputGate.java

```diff
@@ -131,7 +133,18 @@
 	/**
 	 * Setup gate, potentially heavy-weight, blocking operation comparing to just creation.
 	 */
-	public abstract void setup() throws IOException, InterruptedException;
+	public abstract void setup() throws IOException;
+
+	/**
+	 * It is only performed for unaligned checkpoint mode together with internal requesting partitions afterwards.
+	 * Otherwise only {@link #requestPartitions()} is performed for other checkpoint modes.
+	 *
+	 * @param executor the dedicated executor for performing the recovery state for all the internal channels.
+	 * @param reader the dedicated reader for unspilling the respective channel state from stored snapshots.
+	 */
+	public abstract void readRecoveredState(ExecutorService executor, ChannelStateReader reader) throws IOException;
+
+	public abstract void requestPartitions() throws IOException;
```

Comment:
`requestPartitions()` requires similar java doc as `readRecoveredState()` explaining when it should be called.